src/DynamoCore/Linting/LinterManager.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Dynamo.Core;
+using Dynamo.Extensions;
+using Dynamo.Graph.Nodes;
+using Dynamo.Graph.Workspaces;
+using Dynamo.Linting.Interfaces;
+using Dynamo.Linting.Rules;
+using Dynamo.Models;
+
+namespace Dynamo.Linting
+{
+    public class LinterManager : NotificationObject
```

Comment:
Regarding the lifespan and subscription events of LinterManager and Linter rules.
We can have a dispose method here which is going to : 
unsubscribe from the static events and also call dispose on all the rules 
There is a dispose method in DynamoModel where I think we could do the disposal of Linter manager just before the extension manager gets disposed. Unless there is better place to do that ?